utils/misc.mli

```diff
@@ -453,10 +453,32 @@ val debug_prefix_map_flags: unit -> string list
 (** Returns the list of [--debug-prefix-map] flags to be passed to the
     assembler, built from the [BUILD_PATH_PREFIX_MAP] environment variable. *)
 
+val with_geometry :
+  Format.geometry ->
+  (Format.formatter -> 'a -> unit) -> Format.formatter -> 'a -> unit
+(** [with_geometry geometry printer ppf v] runs [printer ppf v] after
+    temporarily setting the geometry to [geometry] -- and reverts
+    to the previous geometry afterwards. *)
+
 val print_if :
+  Format.formatter -> Format.geometry option ->
+  bool ref -> (Format.formatter -> 'a -> unit) -> 'a -> 'a
+(** [print_if ppf geometry flag fmt x] prints [x]
+    with [fmt] on [ppf] if [b] is true.
+    If [geometry] is set, it uses this geometry during printing,
+    see {!with_geometry} above. *)
+
+val dump_if :
   Format.formatter -> bool ref -> (Format.formatter -> 'a -> unit) -> 'a -> 'a
-(** [print_if ppf flag fmt x] prints [x] with [fmt] on [ppf] if [b] is true. *)
+(** Like {!print_if}, with a geometry preset designed for dumping
+    large tree-form intermediate-representation that typically
+    go over default [max_indent] settings. *)
 
+val pretty_print_if :
```

Comment:
Exposing both `print_if` and `pretty_print_if` makes this interface confusing. From the docs and implementation, it's clear that `pretty_print_if` is `print_if` with a default value for the `geometry` arg. This could be made explicit in the types too:

    val print_if : Format.formatter -> ?geometry:Format.geometry option -> ...

Reply:
The reason is as follows:
- `print_if` is the general function that can be used for any sort of printing.. and you have to explicitly choose a geometry (or to use the default geometry).
- `dump_if` is to be used for machine-produced data.
- `pretty_print_if` is to be used for human-produced text.

Maybe the naming is not right, but I think that this API choice makes `print_if` and `pretty_print_if` quite different in the intent their express, and for this reason I don't like your suggestion to merge them together.

Besides, your suggestion also has the property that previous users to `print_if` still type-check fine (we only added an optional parameter). I wanted to make sure that all uses of `print_if` would be flagged as type errors, so that I had to consider whether to use dump_if or pretty_print_if at all callsites, so I intentionally chose to change the `print_if` type. (Now that I did review and change all callsites, it would be safe to change back the type to be compatible with pre-PR, but I think that keeping the change in is the better choice.)

Reply:
I think having all three means there's an unnecessary amount of thought for anyone who needs to print things. If you don't like having an optional argument, maybe the best thing is not to expose `print_if`?

(Your last paragraph explains why you did this while writing this patch, but not why it's still that way. If it's important to you that there be no uses of print_if, then delete it!)

Reply:
One option would be to remove `pretty_print_if` and use `print_if` with the `None` geometry-setting directly. It's fairly uncontroversial that, given no information on the nature of the text, the geometry should be optimized for human readability of "standard" text -- so I guess this is morally the same as `pretty_print_if`? I'll try that and look at the resulting code.